### PR TITLE
website_mass_mailing + website_sale popup

### PR DESCRIPTION
**Pull request by robinlej**

### CODE REVIEW

File: addons/website_mass_mailing/static/src/interactions/popup.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+patch(Popup.prototype, {
+    /**
+     * Prevents the (newsletter) popup to be shown if the user is subscribed.
+     *
+     * @override
+     */
+    canShowPopup() {
+        if (
+            this.el.matches(".o_newsletter_popup")
```
Comment (bso-odoo) [inline addons/website_mass_mailing/static/src/interactions/popup.js]: Why not `.classList.contains(...)` like in the next method ?

File: addons/website_mass_mailing/static/src/interactions/popup.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            && this.el.querySelector("input.js_subscribe_value, input.js_subscribe_email").disabled // js_subscribe_email is kept by compatibility (it was the old name of js_subscribe_value)
```
Comment (bso-odoo) [inline addons/website_mass_mailing/static/src/interactions/popup.js]: What if selector returns null ?

File: addons/website_sale/static/src/interactions/popup.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { patch } from "@web/core/utils/patch";
+import { Popup } from "@website/interactions/popup/popup";
+
+patch(Popup.prototype, {
```
Comment (bso-odoo) [inline addons/website_sale/static/src/interactions/popup.js:4]: Maybe you also want a test about this patch ?

File: addons/website_mass_mailing/static/src/interactions/popup.js (hunk 2 above)
Comment (robinlej) [reply addons/website_mass_mailing/static/src/interactions/popup.js]: ah, translating jQuery is a pain, that's the kind of things that aren't that obvious :sweat_smile:

**Pull request merged**